### ISSUE TITLE
Sass cache issue

### ISSUE THREAD

**Issue by clthck** [markstory/asset_compress#280]
I'm using ScssFilter for compiling sass. It works fine at first, but when I made some changes to scss files and refresh the web page, it didn't update. It worked only after I changed `asset_compress.ini` or `asset_compress.local.ini` file.
Am I missing any config options for resolving this?

**Comment by markstory** [comment]
What does your configuration file look like? Sometimes if the paths end in `/` the mtime detection fails.

**Comment by clthck** [comment]
`asset_compress.ini`
```
[General]
cacheConfig = false
alwaysEnableController = false
themes[] = Red
themes[] = Modern

[css]
paths[] = WEBROOT/assets/stylesheets/*
paths[] = WEBROOT/assets/stylesheets/sections/**
cachePath = WEBROOT/css
timestamp = true

[application.css]
files[] = application.scss
filters[] = ScssFilter
```

`asset_compress.local.ini`
```
[General]
cacheConfig = false
alwaysEnableController = true

[filter_ScssFilter]
sass = C:\Ruby22-x64\bin\sass -r sass-globbing
```

**Comment by markstory** [comment]
Try removing the `/*` from both path settings.

**Comment by clthck** [comment]
Changed the lines to the following:
```
paths[] = WEBROOT/assets/stylesheets/
paths[] = WEBROOT/assets/stylesheets/sections/
```
And tried both with and without trailing slashes, but still not working.

**Comment by clthck** [comment]
Also there's another issue.
If `paths[] = WEBROOT/assets/stylesheets/**/*` works fine for files in any level of subfolders, but not for files at root level. In nodejs, especially using grunt, this includes the root folder itself as well.

For now, I need to write 2 lines to make it work properly, as follows:
```
paths[] = WEBROOT/assets/stylesheets
paths[] = WEBROOT/assets/stylesheets/**/*
```

**Comment by clthck** [comment]
Back to the original issue..
If I change the contents of `application.scss`, it's reflected on page refresh. But if I change the contents of scss files that are imported into `application.scss`, they're not reflected on page refresh.

But it's not like that for javascript files. I'm using Sprockets filter which is included by default.
```
[js]
baseUrl = http://cdn.example.com
timestamp = true
paths[] = WEBROOT/assets/javascripts
paths[] = WEBROOT/assets/javascripts/**/*
paths[] = WEBROOT/assets/components
paths[] = WEBROOT/assets/components/**/*
cachePath = WEBROOT/js
filters[] = Sprockets

[application.js]
files[] = application.js
```

`application.js`
```js
//= require <jquery.min.js>
//= require "base.js"
```

Javascript file contents changes are shown immediately on the browser, either I change contents of `application.js` or `base.js`, or `jquery.min.js`, which is very good.

ScssFilter should work this way too.

**Comment by clthck** [comment]
And `timestamp` is set to `true`, but I don't see timestamp attached in css link or js link when I inspect the page source.

**Comment by clthck** [comment]
Looks like these issues (except timestamp issue) belong to [mini-asset](http://github.com/markstory/mini-asset) rather than [asset_compress](http://github.com/markstory/asset_compress).

**Comment by markstory** [comment]
Timestamps are only applied to statically generated assets. Development builds don't use timestamps.

What do your import statements look like in your scss? I also use ScssFilters and haven't run into stale builds. But I'm probably missing a scenario that you have somewhere.

**Comment by clthck** [comment]
I have forked mini-asset and working on improvement.

**Comment by clthck** [comment]
by the way, I tried to use Uglifyjs filter but it's not working well. I'm on Win 10

**Comment by clthck** [comment]
It throws me the following error:
```
C:\Users\clthck\AppData\Roaming\npm\uglifyjs:4 case `uname` in ^^^^ SyntaxError: Unexpected token case at exports.runInThisContext (vm.js:54:16) at Module._compile (module.js:375:25) at Object.Module._extensions..js (module.js:406:10) at Module.load (module.js:345:32) at Function.Module._load (module.js:302:12) at Function.Module.runMain (module.js:431:10) at startup (node.js:141:18) at node.js:977:3
```

**Comment by clthck** [comment]
I also tried the following command in command prompt:
```
"C:\Program Files\nodejs\node.exe" C:\Users\clthck\AppData\Roaming\npm\uglifyjs - "var a = 5;"
```
It gives me the same error like the above one.

**Comment by markstory** [comment]
Sounds like uglify doesn't work so well on windows. You might have to follow up with the uglify folks.

**Comment by clthck** [comment]
Uglifyjs issue is fixed. I changed uglifyjs path to the correct one as follows:
```
"C:\Program Files\nodejs\node.exe" C:\Users\clthck\AppData\Roaming\npm\node_modules\uglify-js\bin\uglifyjs - "var a = 5;"
```

**Comment by markstory** [comment]
I think this has been resolved as of 1.0.3
